Dev

xlimitan opened the pull request.

Frokolll111 committed:
Commit

Svidinskiy committed:
Создание DTO и контролеров

Svidinskiy committed:
Merge pull request #1 from xlimitan/andrey

Создание DTO и контролеров

Frokolll111 committed:
Merge branch 'dev' into anatoly

# Conflicts:
#	src/main/resources/liquibase/scripts/notification.sql

Frokolll111 committed:
Commit

xlimitan committed:
Создание сущностей и rename DTO

xlimitan committed:
Merge pull request #3 from xlimitan/sam

Создание сущностей и rename DTO

Frokolll111 committed:
Merge branch 'dev' into anatoly

Frokolll111 committed:
Сделал репозитории

Frokolll111 committed:
Сделал репозитории

Frokolll111 committed:
Merge pull request #4 from xlimitan/anatoly

Anatoly

Svidinskiy committed:
Созданы мапперы

Svidinskiy committed:
Merge branch 'dev' into andrey

Svidinskiy committed:
Merge pull request #5 from xlimitan/andrey

Созданы мапперы

Frokolll111 committed:
Сделал 3 маппера

Frokolll111 committed:
Merge pull request #6 from xlimitan/anatoly

Сделал 3 маппера

Svidinskiy committed:
Дополнены мапперы

xlimitan committed:
Создание 3 Mappers

xlimitan committed:
Merge pull request #7 from xlimitan/sam

Создание 3 Mappers

Svidinskiy committed:
Добавлены 5 мапперов

Svidinskiy committed:
Merge remote-tracking branch 'origin/andrey' into andrey

Svidinskiy committed:
Merge pull request #8 from xlimitan/andrey

Andrey

Frokolll111 committed:
Удалил лишний репозиторий , Добавил валидации полей и добавил pluggin

Frokolll111 committed:
Merge pull request #9 from xlimitan/anatoly

Удалил лишний репозиторий , Добавил валидации полей и добавил pluggin

Svidinskiy committed:
Добавлены геттеры вручную

Svidinskiy committed:
Merge pull request #10 from xlimitan/andrey

Добавлены геттеры вручную

xlimitan committed:
Исправил ошибку с pom

xlimitan committed:
Merge pull request #11 from xlimitan/sam

Исправил ошибку с pom

Frokolll111 committed:
Добавил аутентификация с помощью UserDetailsService
, изменил маперы

Frokolll111 committed:
Merge branch 'dev' into anatoly

Frokolll111 committed:
Merge pull request #12 from xlimitan/anatoly

Добавил аутентификация с помощью UserDetailsService

xlimitan committed:
Написание UserService

xlimitan committed:
Написание UserService 2.0.

xlimitan committed:
Merge pull request #13 from xlimitan/sam

Sam

Svidinskiy committed:
Добавлен сервис комментариев

Svidinskiy committed:
Merge pull request #14 from xlimitan/andrey

Добавлен сервис комментариев

Frokolll111 committed:
Добавил сервис объявлений, так же эндпоиты и исключения

Frokolll111 committed:
Merge remote-tracking branch 'origin/anatoly' into anatoly

Frokolll111 committed:
Merge pull request #15 from xlimitan/anatoly

Anatoly

xlimitan committed:
Вернул AuthServiceImpl

xlimitan committed:
Создал реализацию интерфейса UserDetailsManager

xlimitan committed:
Исправил ошибки в MyUserDetailsService

xlimitan committed:
Merge pull request #16 from xlimitan/sam

Sam

xlimitan committed:
Доделал UserController

xlimitan committed:
Оформил контроллеры

xlimitan committed:
Написал JavaDoc дял сервисов

xlimitan committed:
Merge pull request #17 from xlimitan/sam

Sam

xlimitan committed:
Исправил ошибки

xlimitan committed:
Merge pull request #18 from xlimitan/sam

Исправил ошибки

xlimitan committed:
Исправил ошибки со свагером

xlimitan committed:
Merge pull request #19 from xlimitan/sam

Исправил ошибки со свагером

Frokolll111 committed:
Изменил и Добавил возможность фронта сохранить картинку, а потом отобразить ее в профиле

Frokolll111 committed:
Merge branch 'dev' into anatoly

Frokolll111 committed:
Merge pull request #20 from xlimitan/anatoly

Изменил и Добавил возможность фронта сохранить картинку, а потом отоб…

artem-petrosyan commented on src/main/java/ru/skypro/homework/dto/Ads.java:
```diff
@@ -0,0 +1,15 @@
+    @Schema(description = "количество объявлений")
+    private Integer count;
+    @Schema(description = "список объявлений")
+    private List<AdDto> result;
```
```suggestion
    private List<AdDto> results;
```

artem-petrosyan commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,204 @@
+                            description = " Unauthorized"
+                    )})
+    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<AdDto> addAd(@RequestParam MultipartFile image,
+                                        CreateOrUpdateAd createOrUpdateAd,
+                                        Authentication authentication){
```
```suggestion
    public ResponseEntity<AdDto> addAd(@RequestPart("image") MultipartFile image,
                                       @RequestPart("properties") CreateOrUpdateAd createOrUpdateAd,
                                        Authentication authentication){
```

xlimitan committed:
Исправил регистрацию пользователя

xlimitan committed:
Merge pull request #22 from xlimitan/sam

Исправил регистрацию пользователя

xlimitan committed:
Исправление создания комментария 1.0

xlimitan committed:
Merge pull request #23 from xlimitan/sam

Исправление создания комментария 1.0

xlimitan committed:
Исправил оформление контроллеров

xlimitan committed:
Merge pull request #24 from xlimitan/sam

Исправил оформление контроллеров

Frokolll111 committed:
Сделал Удаление и изменения картинки объявления

xlimitan committed:
Исправил оформление контроллеров 2.0

xlimitan committed:
Merge pull request #26 from xlimitan/sam

Исправил оформление контроллеров 2.0

Frokolll111 committed:
Merge branch 'dev' into anatoly

Frokolll111 committed:
Merge pull request #25 from xlimitan/anatoly

Сделал Удаление и изменения картинки объявления

Frokolll111 committed:
Исправил

Frokolll111 committed:
Merge pull request #27 from xlimitan/anatoly

Исправил

artem-petrosyan commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,164 @@
+                            description = "Not found",
+                            content = @Content(schema = @Schema(hidden = true)))})
+    @PatchMapping("/{id}")
+    public ResponseEntity<CreateOrUpdateAd> updateAd (int adId,
```
```suggestion
    public ResponseEntity<CreateOrUpdateAd> updateAd (@PathVariable("id") int adId,
```

artem-petrosyan commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -0,0 +1,111 @@
+
+@RestController
+@RequestMapping("/ads")
+public class CommentController {
```
Нужно добавить строку
`@CrossOrigin(value = "http://localhost:3000")
`

artem-petrosyan commented on src/main/java/ru/skypro/homework/dto/Comments.java:
```diff
@@ -0,0 +1,23 @@
+    @Schema(description = "количество комментариев")
+    private Integer count;
+    @Schema(description = "комментарии")
+    private List<CommentDto> result;
```
```suggestion
    private List<CommentDto> results;
```

Также просмотрите и убедитесь, что у вас больше нет расхождений со спекой

artem-petrosyan commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -47,6 +51,10 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                                         .mvcMatchers(AUTH_WHITELIST)
                                         .permitAll()
                                         .mvcMatchers("/ads/**", "/users/**")
+                                        .hasAnyRole("USER", "ADMIN")
```
```suggestion
                                        .mvcMatchers(AUTH_WHITELIST)
                                        .permitAll()
                                        .mvcMatchers(HttpMethod.GET,"/ads").permitAll()
                                        .mvcMatchers("/ads/**", "/users/**")
                                        .hasAnyRole("USER", "ADMIN")
```
Вот так можно дать доступ к списку объявлений

xlimitan committed:
Исправил ошибки:
- добавления комментария
- изменение объявления
- показ объявлений для всех пользователь

xlimitan committed:
Merge branch 'dev' into sam

xlimitan committed:
Merge pull request #28 from xlimitan/sam

Исправил ошибки:

Frokolll111 committed:
Добавил отображение картинки объявления

Frokolll111 committed:
Merge pull request #29 from xlimitan/anatoly

Добавил отображение картинки объявления

xlimitan committed:
Исправил добавления объявления

xlimitan committed:
Merge branch 'dev' into sam

Frokolll111 committed:
Добавил отображение картинки объявления

Frokolll111 committed:
Merge pull request #30 from xlimitan/anatoly

Добавил отображение картинки объявления

Frokolll111 committed:
Исправил

Frokolll111 committed:
Merge pull request #31 from xlimitan/anatoly

Исправил

Frokolll111 committed:
Исправил

Frokolll111 committed:
Merge pull request #32 from xlimitan/anatoly

Исправил

Frokolll111 committed:
Исправил комментарий

Frokolll111 committed:
Merge pull request #33 from xlimitan/anatoly

Исправил комментарий

Frokolll111 committed:
Исправил

Frokolll111 committed:
Merge pull request #34 from xlimitan/anatoly

Исправил

xlimitan committed:
Сделали присвоение значений счетчикам:
- комментариев
- объявлений

Frokolll111 committed:
Добавил Тесты для AdsController

Frokolll111 committed:
Merge pull request #35 from xlimitan/anatoly

Добавил Тесты для AdsController

xlimitan committed:
Поправил оформление

xlimitan committed:
Merge branch 'dev' into sam

# Conflicts:
#	src/main/java/ru/skypro/homework/dto/AdDto.java

xlimitan committed:
Сделал UserControllerTest

xlimitan committed:
Merge pull request #36 from xlimitan/sam

Sam

Frokolll111 committed:
Добавил Описание

Frokolll111 committed:
Merge pull request #37 from xlimitan/anatoly

Добавил Описание

Svidinskiy committed:
Тест CommentController

Svidinskiy committed:
Merge pull request #38 from xlimitan/andrey

Тест CommentController
